src/Form/Config/CloudDashboardAdminSettings.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\cloud_dashboard\Form\Config;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Class Cloud Admin Settings.
+ */
+class CloudDashboardAdminSettings extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'cloud_dashboard_admin_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return ['cloud_dashboard.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+
+    $config = $this->config('cloud_dashboard.settings');
+
+    $form['custom_urls'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Custom URLs'),
+      '#open' => TRUE,
+    ];
+
+    $form['custom_urls']['oauth2_callback_uri'] = [
+      '#type' => 'url',
+      '#title' => $this->t('Callback URI for OAuth2'),
+      '#default_value' => $config->get('oauth2_callback_uri'),
+      '#description' => $this->t('If you leave this field blank, the callback URI of Comsumer plugin on the same server as the Cloud Dashboard is used.'),
+      '#attributes'    => ['placeholder' => 'e.g. http://example.com/clouds/dashboard/callback'],
+    ];
+
+    $form['custom_urls']['oauth2_client_id'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Client ID for OAuth2'),
+      '#default_value' => $config->get('oauth2_client_id'),
+      '#description' => $this->t('If you leave this field blank, the client ID of Comsumer plugin on the same server as the Cloud Dashboard is used.'),
```

Comment:
@RyoYamashita765 This can be:

```suggestion
      '#description' => $this->t('If you leave this field blank, the client ID will be the same as the consumer plugin at the server of Cloud Dashboard.'),
```